std/datetime/date.d

```diff
@@ -7280,35 +7280,42 @@ public:
     static Date fromISOString(S)(in S isoString) @safe pure
         if (isSomeString!S)
     {
-        import std.algorithm.searching : all, startsWith;
-        import std.ascii : isDigit;
-        import std.conv : to;
+        import std.algorithm.searching : startsWith;
+        import std.conv : to, text, ConvException;
         import std.exception : enforce;
-        import std.format : format;
         import std.string : strip;
 
-        auto dstr = to!dstring(strip(isoString));
+        auto str = isoString.strip;
 
-        enforce(dstr.length >= 8, new DateTimeException(format("Invalid ISO String: %s", isoString)));
+        enforce!DateTimeException(str.length >= 8, text("Invalid ISO String: ", isoString));
 
-        auto day = dstr[$-2 .. $];
-        auto month = dstr[$-4 .. $-2];
-        auto year = dstr[0 .. $-4];
+        int day, month, year;
+        auto yearStr = str[0 .. $ - 4];
 
-        enforce(all!isDigit(day), new DateTimeException(format("Invalid ISO String: %s", isoString)));
-        enforce(all!isDigit(month), new DateTimeException(format("Invalid ISO String: %s", isoString)));
+        try
+        {
+            // using conversion to uint plus cast because it checks for +/-
+            // for us quickly while throwing ConvException
+            day = cast(int) to!uint(str[$ - 2 .. $]);
+            month = cast(int) to!uint(str[$ - 4 .. $ - 2]);
```

Comment:
All of these `to!` calls (and the ones below) now autodecode, because `str` is no longer guaranteed to be a `dstring`, and so they risk throwing a `UTFException`. And I don't think that `representation` will work in this case. Try `byCodeUnit`, but I don't know if the bug preventing std.conv working with `byCodeUnit` was fixed yet. If not, given the clear speed-up, we can temporarily add a catch for `UTFException` so that we can rethrow a `DateTimeException`, but ultimately, none of this code should need to be doing anything with auto-decoding (at least outside of the error messages, which unfortunately do risk it with `text`, but I don't know what to do about that).